src/node_crypto.cc

```diff
@@ -2773,6 +2775,38 @@ void SSLWrap<Base>::GetSharedSigalgs(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+template <class Base>
+void SSLWrap<Base>::ExportKeyingMaterial(
+                                      const FunctionCallbackInfo<Value>& args
+                                      ) {
+  CHECK(args[0]->IsInt32());
+  CHECK(args[1]->IsString());
+
+  Base* w;
+  ASSIGN_OR_RETURN_UNWRAP(&w, args.Holder());
+  Environment* env = w->ssl_env();
+
+  uint32_t olen = args[0].As<Uint32>()->Value();
+  node::Utf8Value label(env->isolate(), args[1].As<String>());
+
+  AllocatedBuffer out = env->AllocateManaged(olen);
+
+  if (SSL_export_keying_material(w->ssl_.get(),
+              reinterpret_cast<unsigned char*>(out.data()),
+              olen,
+              *label,
+              label.length(),
+              nullptr,  // the context parameter is optional
+              0,
+              0) != 1) {
```

Comment:
Style: arguments should line up with the first one.

Substance: `SSL_export_keying_material()` leaves an error message on openssl's error stack that you should throw instead. See `CryptoErrorVector` or `ThrowCryptoError()`; the second one is probably easiest to use:
```c++
return ThrowCryptoError(env, ERR_get_error(), "SSL_export_keying_material");
```

Reply:
thanks, fixed this